src/libAtomVM/nifs.c

```diff
@@ -3494,11 +3494,11 @@ static term nif_erlang_ref_to_list(Context *ctx, int argc, term argv[])
 
     term t = argv[0];
     VALIDATE_VALUE(t, term_is_reference);
+    size_t max_len = term_is_external(t) ? EXTERNAL_REF_AS_CSTRING_LEN : LOCAL_REF_AS_CSTRING_LEN;
 
-    char buf[REF_AS_CSTRING_LEN];
-    int str_len = term_snprint(buf, REF_AS_CSTRING_LEN, t, ctx->global);
+    char buf[max_len];
```

Comment:
I suggest avoiding VLA as much as possible, and I would rather just use the max between the 2.
It seems that they are so bad that in Linux kernel they worked quite a lot to remove all of them.

Reply:
I'm not sure Linus' hatred for VLA is the best argument here :)

In this case it means usually allocating 40 more bytes on the stack, but the performance penalty should be negligible and I agree this yields a cleaner code.